velodyne_laserscan/src/velodyne_laserscan.cpp

```diff
@@ -179,9 +179,7 @@ void VelodyneLaserScan::recvCallback(const sensor_msgs::msg::PointCloud2::Shared
       scan->intensities.resize(kSize);
 
       for (sensor_msgs::PointCloud2ConstIterator<float> it(*msg, "x"); it != it.end(); ++it) {
-        const uint16_t r =
-          (static_cast<const uint16_t>(it[5]) << 8) |
-          static_cast<const uint16_t>(it[6]);
+        const uint16_t r = *(reinterpret_cast<const uint16_t *>(&it[5]));
```

Comment:
So, if I'm reading this correctly, `it[5]` returns a float value (reference, I guess, but that doesn't make much difference here).  We then take the address of that, so we now have a `float *`.  Then we `reinterpret_cast` that to a `uint16_t *`, and then we dereference that and store it into the `uint16_t`.  But isn't that meaningless?  That is, casting a `float *` to a `uint16_t *` just gives you some of the bits of the mantissa (or however the float is laid out bitwise, I forget the details).  Am I missing something here?

Reply:
Your reasoning is correct. The thing is, `ring` actually is a `uint16_t` and it is layed out in memory as such. Reading it as a `float` is what is meaningless, for the reasons you've described. I presume it was written this way to spare another iterator (a performance boost?) and I didn't want to change that.

Reply:
Oh!  Wow, that is subtle.  Thanks for the explanation, that is helpful.  Would you mind adding a comment above this code basically explaining that?  I think it will save confusion for someone in the future.  Otherwise, looks good to me.

Reply:
Done in https://github.com/ros-drivers/velodyne/pull/322/commits/12f6263e9ec279ccb8b4a2e48bddc2b63774068a.